### PR TITLE
HAL-1271: fix handling of groups-properties

### PR DESCRIPTION
**Pull request by michpetrov**
HAL: [HAL-1271](https://issues.jboss.org/browse/HAL-1271)
JBEAP: [JBEAP-8162](https://issues.jboss.org/browse/JBEAP-8162)

### CODE REVIEW

**Comment by hpehl** [comment]
@claudio4j can you please review and merge? Thanks.

File: gui/src/main/java/org/jboss/as/console/mbui/widgets/ComplexAttributeForm.java [hunk 1]
```diff
@@ -70,7 +77,17 @@ public ComplexAttributeForm(String attributeName, SecurityContext securityContex
 
     private ResourceDescription getAttributeDescription() {
         ModelNode desc = new ModelNode();
-        desc.get("attributes").set(resourceDescriptionDelegate.get("attributes").get(attributeName).get("value-type"));
+        desc.get(ATTRIBUTES).set(resourceDescriptionDelegate.get(ATTRIBUTES).get(attributeName).get(VALUE_TYPE));
+        boolean nillable = resourceDescriptionDelegate.get(ATTRIBUTES).get(attributeName).hasDefined(NILLABLE) ?
+                resourceDescriptionDelegate.get(ATTRIBUTES).get(attributeName).get(NILLABLE).asBoolean() : false;
+        if (nillable) {
```
Comment (claudio4j) [inline gui/src/main/java/org/jboss/as/console/mbui/widgets/ComplexAttributeForm.java]: This is not a general case, to set the nested attributes as required=true if the parent attribute is required=true, one example is subsystem=elytron/key-store resource, the credential-reference is an attribute (required=true) that contains 4 attributes all of them required=false, either the user can set store or clear-text/type/alias. Other example is identity-mapping OBJECT attribute of elytron/ldap-realm resource. That said, I request to remove this modification to ComplexAttributeForm.java. Thanks.

Comment (michpetrov) [inline gui/src/main/java/org/jboss/as/console/mbui/widgets/ComplexAttributeForm.java]: As described in the comment this is doing the opposite i.e. setting nillable=true (required=false) if the parent object is nillable. If it is not desirable to do it this way there still has to be a way to let the user set the whole object to null even if some of its nested attributes are required.

Comment (claudio4j) [inline gui/src/main/java/org/jboss/as/console/mbui/widgets/ComplexAttributeForm.java]: Ah, sorry my bad, confused nillable with required. I will merge it.

**Pull request merged**